src/hotspot/share/gc/parallel/psParallelCompact.cpp

```diff
@@ -2137,13 +2137,9 @@ size_t PSParallelCompact::next_src_region(MoveAndUpdateClosure& closure,
   }
 
   if (src_region_ptr < top_region_ptr) {
-    // The next source region is in the current space.  Update src_region_idx
-    // and the source address to match src_region_ptr.
+    // Found the first non-empty region in the same space.
     src_region_idx = sd.region(src_region_ptr);
-    HeapWord* const src_region_addr = sd.region_to_addr(src_region_idx);
-    if (src_region_addr > closure.source()) {
-      closure.set_source(src_region_addr);
-    }
+    closure.set_source(sd.region_to_addr(src_region_idx));
```

Comment:
Just to make sure I understand: the only change here is the removal of the condition `src_region_addr > closure.source()` because at worst we can set the same value into `closure._source` anyway, and the additional check is kind of superfluous.